src/store/CrosshairStore.ts

```diff
@@ -97,17 +99,24 @@ class CrosshairStore {
     };
 
     onMouseMove = (dx: number, dy: number, epoch: number, quote: string) => {
+        if (this.cachedDx === dx && this.cachedDy === dy) {
+            return;
+        }
+        this.cachedDx = dx;
+        this.cachedDy = dy;
         if (this.hoverOnScreen === false) {
             this.isOverChartContainer = true;
-            this.updateVisibility(true);
             this.hoverOnScreen = true;
         }
-
         if (!this.isOverChartContainer) return;
 
+        if (this.layoutCrosshair) {
+            this.setCrosshairState(this.layoutCrosshair);
+            this.layoutCrosshair = 0;
```

Comment:
Why do we need to call a function  `this.setCrosshairState`, if then on line 115 we set it to zero?
Literally it seems like:
```
this.layoutCrosshair = this.layoutCrosshair;
this.layoutCrosshair = 0;
```
Sorry I didn't get the idea :( Please, ignore if it's not relevant

Reply:
So layoutCrosshair is crosshair value that is set up from the template . Once the template is clicked . It gets the crosshair of the template into layoutCrosshair and not on the setCrosshairState(the actual crosshair state) . It only set it up on   onmousemove  function.
The above code . It checks if there is a layoutCrosshair state . If yes it sets up the actual crosshair state (inside setCrosshairState function) from layoutCrosshair and then make its value 0 until a new template is set up.

Reply:
Thank you for explanation